From c0cb8f17d8300790a5e65b3f2d51c7dd66b53b85 Mon Sep 17 00:00:00 2001
From: Erez Shaul <eshaul@habana.ai>
Date: Tue, 08 Oct 2024 10:44:55 +0300
Subject: [PATCH] [SW-190033] boot_desc_init: do not assume struct size

SPI offset to read all descriptors in not necessarily the size of
hl_fw_boot_ptrs. it is explicitly stored in offset2array feild.

Change-Id: I7a94757bf8fe3c09e0b5caa33513abadeed8d502
Signed-off-by: Erez Shaul <eshaul@habana.ai>
---

diff --git a/lib/common/boot_desc.c b/lib/common/boot_desc.c
index 5b11c49..1ae0341 100644
--- a/lib/common/boot_desc.c
+++ b/lib/common/boot_desc.c
@@ -145,7 +145,7 @@
 	}
 
 	/* Read all the descriptors offsets */
-	spi_off += sizeof(struct hl_fw_boot_ptrs);
+	spi_off += boot_ptrs.offset2array;
 	spi_flash_copy(desc_offs, spi_off, sizeof(u32) * boot_ptrs.count, "DESC OFFS");
 
 	/* Read all descriptors */
